drivers/clock_control/nrf_power_clock.c

```diff
@@ -149,34 +200,37 @@ static int clock_async_start(struct device *dev,
 	ref = ++clk_data->ref;
 	irq_unlock(key);
 
-	if (clk_data->started) {
-		if (data) {
-			data->cb(dev, data->user_data);
-		}
-	} else {
-		if (ref == 1) {
-			bool do_start;
-
-			do_start =  (config->start_handler) ?
-					config->start_handler(dev) : true;
-			if (do_start) {
-				nrf_clock_task_trigger(NRF_CLOCK,
-						       config->start_tsk);
-				DBG(dev, "Triggered start task");
-			} else if (data) {
-				data->cb(dev, data->user_data);
-			}
-		}
-
+	if (data) {
 		/* if node is in the list it means that it is scheduled for
 		 * the second time.
 		 */
-		if (data) {
-			if (is_in_list(&clk_data->list, &data->node)) {
-				return -EALREADY;
-			}
+		if (is_in_list(&clk_data->list, &data->node)) {
+			return -EALREADY;
+		}
 
-			sys_slist_append(&clk_data->list, &data->node);
+		clock_irqs_disable();
+		started = clk_data->started;
+		if (!clk_data->started) {
```

Comment:
Use of `started` (or `already_started` as suggested above) here would be a bit more appropriate, to indicate that this is a counterpart of the below code that calls `data->cb` directly.

Reply:
done